test/e2e/storage/persistent_volumes-local.go

```diff
@@ -252,6 +252,39 @@ var _ = SIGDescribe("PersistentVolumes-local [Feature:LocalPersistentVolumes] [S
 		})
 	})
 
+	Context("when pod using local volume with non-existant path", func() {
+		It("should not be able to mount", func() {
+			for _, testVolType := range LocalVolumeTypes {
+				By(fmt.Sprintf("local-volume-type: %s", testVolType))
+				testVol := &localTestVolume{
+					node:            config.node0,
+					hostDir:         "/non-existent/location/nowhere",
+					localVolumeType: testVolType,
+				}
+				By("Creating local PVC and PV")
+				createLocalPVCPV(config, testVol)
+				_, pod1Err := createLocalPod(config, testVol)
+				Expect(pod1Err).To(HaveOccurred())
+			}
+		})
+	})
+	Context("when pod has nodeName set to a different node than the PV node affinity", func() {
+		It("should not be able to mount", func() {
+			for _, testVolType := range LocalVolumeTypes {
+				var testVol *localTestVolume
+				By(fmt.Sprintf("local-volume-type: %s", testVolType))
+				testVol = setupLocalVolumePVCPV(config, testVolType)
+				wrongNode := fmt.Sprintf("%s-%s", config.node0.Name, "wrong")
+				pod := makeLocalPodWithNodeSelector(config, testVol, wrongNode)
```

Comment:
There's also the old way of setting Pod.nodeName.  Could you add that as a 2nd test case?

Reply:
Would you be ok if I create second func makeLocalPodWithNodeSelectorOld or should I pass an argument got new or old node selector?

Reply:
second function is fine